server/screens.js

```diff
@@ -56,9 +109,17 @@ module.exports.add = function(socket) {
 
 		// If screen has not cited a specific ID, assign one
 		if (!data.id || !parseInt(data.id, 10)) {
-			data.id = Math.random() * 99999 | 0;
+			data.id = generateID();
+		}
+
+		const thereIsAConflict = checkForConflictingScreens(data);
+
+		if(thereIsAConflict){
+			decideWhichScreenGetsToKeepAnID(data, assignedIDs.filter(s => { return s.id == data.id; } )[0] );
```

Comment:
We can replace the `filter()[0]` with the [find](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find) method